kolibri/core/assets/src/core-actions.js

```diff
@@ -54,9 +126,214 @@ function togglemodal(store, bool) {
   }
 }
 
+
+/**
+ * Create models to store logging information
+ * To be called on page load for content renderers
+ */
+function initContentSession(store, Kolibri, channelId, contentId, contentKind) {
+  const ContentSessionLogResource = Kolibri.resources.ContentSessionLogResource;
+  const ContentSummaryLogResource = Kolibri.resources.ContentSummaryLogResource;
+
+  /* Create summary log iff user exists */
+  if (store.state.core.session.user_id) {
+     /* Fetch collection matching content and user */
+    const summaryCollection = ContentSummaryLogResource.getCollection({
+      content_id: contentId,
+      user: store.state.core.session.user_id,
+    });
+    summaryCollection.fetch().then(summary => {
+      /* If a summary model exists, map that to the state */
+      if (summary.length > 0) {
+        store.dispatch('SET_LOGGING_SUMMARY_STATE', _contentSummaryLoggingState(summary[0]));
+      } else {
+        /* If a summary model does not exist, create default state */
+        store.dispatch('SET_LOGGING_SUMMARY_STATE', _contentSummaryLoggingState({
+          pk: null,
+          start_timestamp: new Date(),
+          completion_timestamp: null,
+          end_timestamp: null,
+          progress: 0,
+          time_spent: 0,
+          extra_fields: '{}',
+          time_spent_before_current_session: 0,
+          progress_before_current_session: 0,
+        }));
+
+        const summaryData = Object.assign({
+          channel_id: channelId,
+          content_id: contentId,
+          kind: contentKind,
+        }, _contentSummaryModel(store));
+
+        /* Save a new summary model and set id on state */
+        const summaryModel = ContentSummaryLogResource.createModel(summaryData);
+        summaryModel.save().then((newSummary) => {
+          store.dispatch('SET_LOGGING_SUMMARY_ID', newSummary.pk);
+        });
+      }
+    });
+  }
+
+  /* Set session log state to default */
+  store.dispatch('SET_LOGGING_SESSION_STATE', _contentSessionLoggingState({
+    pk: null,
+    start_timestamp: new Date(),
+    end_timestamp: null,
+    time_spent: 0,
+    progress: 0,
+    extra_fields: '{}',
+  }));
+
+  const sessionData = Object.assign({
+    channel_id: channelId,
+    content_id: contentId,
+    kind: contentKind,
+  }, _contentSessionModel(store));
```

Comment:
Similarly here -- could just get pulled from the store, within `_contentSessionModel`.